pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -362,14 +369,90 @@ func (c *AntreaIPAMController) processNextStatefulSetWorkItem() bool {
 	return true
 }
 
+func (c *AntreaIPAMController) updateIPPoolCounters(poolName string) error {
+	ipPool, err := c.ipPoolLister.Get(poolName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to retrieve IPPool %s, error: %v", poolName, err)
+	}
+
+	allocator, err := poolallocator.NewIPPoolAllocator(ipPool.Name, c.crdClient, c.ipPoolLister)
+
+	if err != nil {
+		return fmt.Errorf("failed to initialize allocator for IPPool %s, error: %v", poolName, err)
+	}
+
+	// Total is fetched from allocator as here are trapped changes to CRD, e.g addition of new IPRange
+	total := allocator.Total()
+
+	// Used is gathered from IP allocation status within the CRD - as it can be set by each one of the agents
+	used := len(ipPool.Status.IPAddresses)
+
+	// If update has no effect, exit
+	if ipPool.Status.Usage.Used == used && ipPool.Status.Usage.Total == total {
+		return nil
+	}
+	ipPool.Status.Usage.Used = used
```

Comment:
Objects returned here must be treated as read-only as they are shared among all consumers.
```
ipPoolToUpdate := ipPool.DeepCopy()
ipPoolToUpdate.Status.Usage.Used = used
...
```